src/module/types.rs

```diff
@@ -29,6 +29,11 @@ impl ModuleTypes {
         self.arena.iter().map(|(_, f)| f)
     }
 
+    /// Delete a type from this module
+    pub fn delete(&mut self, ty: TypeId) {
```

Comment:
Worth adding a comment about references to the deleted type, like the other `delete` methods.